File: 5kyu/human_readable_time.py
def make_readable(seconds):
    (ss1, ss2) = divmod(seconds, 60)
    (hh, mm2)  = divmod(ss1, 60)
    return ":".join(map(trans, [hh, mm2, ss2]))

def trans(x):
    return "0" + str(x) if x < 10 else str(x)

File: 5kyu/human_readable_time.hs
import Data.List

humanReadable :: Int -> String
humanReadable x = intercalate ":" $ map myshow [hh, mm2,ss2]
                  where (ss1,ss2) = x `divMod` 60
                        (hh,mm2) = ss1 `divMod` 60

myshow x | x < 10    = '0':show x
         | otherwise = show x